docs/cookbook/named_entity_recognition.md

```diff
@@ -0,0 +1,269 @@
+# Named Entity Recognition
+
+This guide demonstrates techniques to perform Named Entity Recognition (NER) using Large Language Models (LLMs) with various levels of nested entity recognition. We'll use Groq's llama-3.1-8b-instant model, but you can adapt this approach to other models with similar capabilities.
+
+??? tip "Mirascope Concepts Used"
+
+    - [Calls](../learn/calls.md)
+    - [Response Model](../learn/response_models.md)
+    - [Prompt Templates](../learn/prompt_templates.md)
+
+## Background
+
+Named Entity Recognition is a subtask of information extraction that seeks to locate and classify named entities in text into predefined categories such as person names, organizations, locations, etc. LLMs have revolutionized NER by enabling more context-aware and hierarchical entity recognition, going beyond traditional rule-based or statistical methods.
+
+## Setup
+
+First, ensure you have the necessary packages installed and API keys set:
+
+```bash
+pip install mirascope
```

Comment:
nit: should be `pip install "mirascope[groq]"`

Reply:
Changed it !